compiler/rustc_interface/src/interface.rs

```diff
@@ -36,8 +36,8 @@ pub type Result<T> = result::Result<T, ErrorGuaranteed>;
 /// Can be used to run `rustc_interface` queries.
 /// Created by passing [`Config`] to [`run_compiler`].
 pub struct Compiler {
-    pub sess: Session,
-    pub codegen_backend: Box<dyn CodegenBackend>,
+    pub sess: Lrc<Session>,
+    pub codegen_backend: Lrc<dyn CodegenBackend>,
```

Comment:
Where exactly does the requirement for turning `Box<dyn CodegenBackend>` into `Lrc<dyn CodegenBackend>` come from?

Reply:
It's used in the `task` which calls `link`.